core/primitives/src/epoch_manager.rs

```diff
@@ -132,12 +132,16 @@ impl AllEpochConfig {
     pub fn for_protocol_version(&self, protocol_version: ProtocolVersion) -> EpochConfig {
         let mut config = self.genesis_epoch_config.clone();
 
+        Self::config_mocknet(&mut config, &self.chain_id, protocol_version);
+
         Self::config_stateless_net(&mut config, &self.chain_id, protocol_version);
 
         if !self.use_production_config {
             return config;
         }
 
+        Self::config_validator_selection(&mut config, protocol_version);
```

Comment:
I think this should go before the `use_production_config` check. Otherwise shuffling wouldn't get enabled unless this flag is set.

Reply:
Done